### DISCUSSION TITLE
How to Pass an Environment Variable to the Docker Container

### DISCUSSION THREAD

**Discussion by ZuluPilgrim** [pglombardo/PasswordPusher#951] [Q&A] [answered]
I tried using sudo docker run -e PWPUSH_MASTER_KEY=0c110f7f9d93d2122f36debf8a24bf835f33f248681714776b336849b801f693 (with a key from the helper tool) but got the following error

`"docker run" requires at least 1 argument.`

As I meet this requirement, I am not sure what I am doing wrong. Is there a better way to do this or how do I correct the above?

**Comment by pglombardo** [comment, marked as answer]
If you have a lot of environment variables to set, you can use a [Docker environment file](https://github.com/pglombardo/PasswordPusher/blob/master/containers/docker/pwpush-docker-env-file).

So either do modifications via environment variables or a custom `settings.yml`.

The specific problem with the command you posted above is that the `-e` option is used when launching a container.  There was no container specified on the command line.

Here's another command line example with an environment variable:

```sh
docker run -d --env PWP__DEFAULT_LOCALE=nl -p "5100:5100" pglombardo/pwpush-ephemeral:release
```

I know this is a lot of information and Docker isn't the easiest software in the world.  But after some time, I'm sure you'll get it.  Once Docker is understood and is working, it's extremely reliable.

**Reply by ZuluPilgrim** [reply]
Ok, so I can simply add it to my custom settings.yml file. I thought I was limited to just what was originally in it. As I have that working, I will put it there.

Thanks

**Reply by pglombardo** [reply]
Environment variables work when you have a few but `settings.yml` is better for many changes.  You can mix and match.  Environment variables override any settings in `settings.yml`.

**Reply by ZuluPilgrim** [reply]
Considering

```
sudo docker run -d --env PWPUSH_MASTER_KEY=98c3333333333333 --mount type=bind, source=/home/sysadmin/pwdpusher/settings.yml,target=/opt/PasswordPusher/config/settings.yml -p "5100:5100" -v /home/sysadmin/pwdpusher/logos:/opt/PasswordPusher/public/logos pglombardo/pwpush-ephemeral:release
``` 

To put PWPUSH_MASTER_KEY=98c3333333333333 into Settings, do I just change it to PWPUSH_MASTER_KEY: 98c3333333333333

**Reply by pglombardo** [reply]
`PWPUSH_MASTER_KEY` is only configurable via environment variable currently.

**Comment by CBB3810** [comment]
For the life of me I can't get the solar theme to work I just downloaded my docker image on Friday.
docker run -d -p "5100:5100" -v /opt/pwpush/logos:/opt/PasswordPusher/public/logos --mount type=bind,source=/opt/pwpush/yaml/settings.yml,target=/opt/PasswordPusher/config/settings.yml --env PWP__THEME=solar pwpush-local

Do I need to put the theme directive in a different place on the command string?

**Reply by CBB3810** [reply]
Please allow me to remove my foot from my mouth. I apologize in advance for the post. I see your documentation now as to how to apply a theme. I didn't see it prior to posting.

**Reply by pglombardo** [reply]
Thanks for posting @CBB3810.  For everyone else coming here, when configuring a theme using `PWP__THEME`, you also have to set `PWP_PRECOMPILE=true` to force asset recompilation.  See the documentation here: https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#configuring-a-theme

> Make sure to set both PWP__THEME and PWP_PRECOMPILE for the selected theme to work.
